pulp_container/app/registry_api.py

```diff
@@ -836,15 +894,25 @@ def put(self, request, path, pk=None):
             ca.save()
         except IntegrityError:
             pass
+        # both docker/oci format should contain layers, digest, media_type, size
         layers = content_data.get("layers")
-        blobs = []
+        blobs = {}
         for layer in layers:
-            blobs.append(layer.get("digest"))
-        blobs_qs = models.Blob.objects.filter(digest__in=blobs)
+            blobs[layer.get("digest")] = layer.get("mediaType")
+        blobs_qs = models.Blob.objects.filter(digest__in=blobs.keys())
         thru = []
         for blob in blobs_qs:
+            # ensure there are no foreign layers
+            blob_media_type = blobs[blob.digest]
+            if blob_media_type not in (
+                models.MEDIA_TYPE.REGULAR_BLOB,
+                models.MEDIA_TYPE.REGULAR_BLOB_OCI,
+            ):
+                raise BlobInvalid(digest=blob.digest)
+            blob.media_type = blob_media_type
             thru.append(models.BlobManifest(manifest=manifest, manifest_blob=blob))
         models.BlobManifest.objects.bulk_create(objs=thru, ignore_conflicts=True, batch_size=1000)
+        models.Blob.objects.bulk_update(objs=blobs_qs, fields=["media_type"])
```

Comment:
The same here. Is there a way to have the Blob content type available when creating the blob?